mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -1753,11 +1757,66 @@ void CMomentumPlayer::TimerCommand_Reset()
             // MOM_TODO do a trace downwards from the top of the trigger's center to touchable land, teleport the player there
             Teleport(&pCurrentZone->WorldSpaceCenter(), nullptr, &vec3_origin);
         }
+        return;
+    }
+
+    const Vector *pVec = nullptr;
+    const QAngle *pAng = nullptr;
+
+    if (stage == 1)
+    {
+        // Stage 1 is the start zone. If the timer has a mark, we use it
+        const auto pStartMark = GetStartMark(track);
+        if (pStartMark)
+        {
+            pVec = &pStartMark->pos;
+            pAng = &pStartMark->ang;
+        }
         else
         {
-            Warning("Cannot reset, you have no current zone!\n");
+            // If no mark was found, we teleport to the center of the start trigger
+            CBaseEntity *pEnt = g_pMomentumTimer->GetStartTrigger(track);
+            if (pEnt)
+            {
+                pVec = &pEnt->GetAbsOrigin();
+
+                if (track != m_Data.m_iCurrentTrack)
+                {
+                    ResetRunStats();
+                    m_Data.m_iCurrentTrack = track;
+                }
+            }
         }
     }
+    else
+    {
+        // Every other index is probably a stage (What about < 1 indexes? Mappers are weird and do "weirder"
+        // stuff so...)
+        CTriggerStage *pStage = nullptr;
+
+        while ((pStage = static_cast<CTriggerStage *>(gEntList.FindEntityByClassname(pStage, "trigger_momentum_timer_stage"))) != nullptr)
+        {
+            if (pStage && pStage->GetZoneNumber() == stage && pStage->GetTrackNumber() == track)
```

Comment:
`pStage` is only used, assigned, and null checked in the while loop:
```suggestion
        while ((auto pStage = static_cast<CTriggerStage *>(gEntList.FindEntityByClassname(pStage, "trigger_momentum_timer_stage"))) != nullptr)
        {
            if (pStage->GetZoneNumber() == stage && pStage->GetTrackNumber() == track)
```

Reply:
Left this as is since the assignment uses itself, which depends on it being declared. Also is now the last code in the method.